### ISSUE TITLE
Not-used column causes dplyr::mutate() to throw exception.

### ISSUE THREAD

**Issue by JohnMount** [tidyverse/dplyr#3069]
A `dplyr::mutate` constructing a new column errors-out due the presence of an irrelevant column of type `raw`.

``` r
suppressPackageStartupMessages(library("dplyr"))
packageVersion("dplyr")
#> [1] '0.7.2.9000'

d <- data.frame(q = 1:3, 
                z = as.raw(1:3))
d %>% 
  mutate(., constant=1.0)
#> Error in mutate_impl(.data, dots): Column `z` is of unsupported type raw vector
```

**Comment by lionel-** [comment]
cf PR #2952

**Comment by JohnMount** [comment]
Yes, but I am not attempting to work with the `raw` column.  `dplyr` just fails to ignore it.

**Comment by lionel-** [comment]
But we won't have to ignore it once we support it.

**Comment by romainfrancois** [comment]
@JohnMount this works now in the `raw` branch, and #2952 PR. Can you test it on your side ?

```
> d <- data.frame(q = 1:3, 
+                 z = as.raw(1:3))
> d %>% 
+   mutate(., constant=1.0)
  q  z constant
1 1 01        1
2 2 02        1
3 3 03        1
```

**Comment by romainfrancois** [comment]
Fixed in #2952
